Makefile

```diff
@@ -25,24 +25,17 @@ all: build test
 test:
 	cd $(ROOT_DIR) && cargo test --all-targets --all-features
 	cd $(ROOT_DIR) && cargo fmt -- --check
-	cd $(ROOT_DIR) && cargo clippy
+	cd $(ROOT_DIR) && cargo clippy -- -D warnings -W clippy::pedantic
 
 test-nix:
 	cd $(ROOT_DIR) && NIX_PATH=nixpkgs=channel:nixos-unstable nix flake check --extra-experimental-features "nix-command flakes" --verbose
 	cd $(ROOT_DIR) && NIX_PATH=nixpkgs=channel:nixos-unstable nix build --extra-experimental-features "nix-command flakes" --verbose
 
+# deprecated
 test-full: test
-	cargo clippy --\
-		-D warnings\
-		-W clippy::pedantic\
-		-A clippy::must_use_candidate\
-		-A clippy::cast_precision_loss\
-		-A clippy::cast_possible_truncation\
-		-A clippy::cast_possible_wrap\
-		-A clippy::cast_sign_loss\
-		-A clippy::mut_mut
+	@printf '\n\033[38;5;9mNote: \"make test-full\" is deprecated. You can use \"make test\" instead with the same functionality\n'
 
-test-full-nix: test-full test-nix
+test-full-nix: test test-nix
```

Comment:
Do we want to rename this to `test-nix` as opposed to `test-full-nix`?

Reply:
I already thought about that, but I didn’t rename it because it would mean another breaking change. I guess no one is using that in an automated script anyways so I can deprecate it too.

Reply:
After looking at the code, I noticed that the nix test itself is already named `test-nix`. So do you want `test-nix` to also run `test` or just deprecate `test-full-nix` with the note to run both as a replacement?

Reply:
Ah yeah; I can see the use case where we would want to keep both `test-nix` and `test-nix-full` separate. I think we can leave it unless there are concerns.